sections/advanced/email-testing.md

```diff
@@ -0,0 +1,213 @@
+Typically email testing involves validating email fields (from, to, cc, subject, attachments), html content and links in the email. Email services also allow spam checks and visual checks.
+The core goal is to enable the last mile of end to end testing, to enable a typical webapp to be test from start to finish.
+
+That being stated, email testing is a fundamental enabler for SaaS test architectures by enabling stateless tests that can scale; tests that independently handle their state and can be executed by n number of entities at the same time. 
+Check out the topic [**Test States**](./sections/advanced/test-states.md) for further discussion on the topic.
+
+<br/><br/>
+
+
+## Foreword
+
+If you are using [gmail tricks](https://www.idownloadblog.com/2018/12/19/gmail-email-address-tricks/) or [AWS Simple Email Service](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-simulator.html) and these use cases are satisfactory for your test needs without any side effects, only topic 1 might be of interest to you.
+
+There are plenty of [email testing solutions](https://www.g2.com/search/products?max=10&query=email+testing) available, and combinations of test frameworks that integrate with them.
+For the code snippets and working examples we will be using [Cypress](https://www.cypress.io/) and [Mailosaur](https://mailosaur.com/), but the ideas should generally apply to any tuple of email services and test automation frameworks.
+ 
+When using Cypress with Mailosaur, there are 2 test-development approaches:
+
+* Implement [Mailosaur API](https://docs.mailosaur.com/reference) using Cypress api testing capabilities using `cy.request()` or `cy.api()`. Utilize plugins and helper utilities to construct test suites.
+
+* Utilize [Mailosaur's Node package](https://www.npmjs.com/package/mailosaur) and implement them using [`cy.task()`](https://docs.cypress.io/api/commands/task.html#Syntax) which allows to run node within Cypress.
+
+> check out [cypress-mailosaur-recipe](https://github.com/muratkeremozcan/cypressExamples/tree/master/cypress-mailosaur) for a working example with these approaches. Note that you will have to start a new Mailosaur trial account and replace environment variables for yourself.
+
+<br/><br/>
+
+
+## (1) Explanation & Code sample - Enabling stateless, scalable tests
+
+Stateless tests that can scale are a necessity in any modern web application testing. We want tests that independently handle their state and tests that can be executed by *n* number of entities at the same time. 
+
+
+While testing SaaS applications, which generically have Subscriptions, Users, Organizations (ex: [Cypress Dashboard Service](https://dashboard.cypress.io/organizations)) a lot of the end to end workflows can rely on having unique users. Elsewise only one test execution can happen at a time and they clash with other simultaneous test executions. This constraint reduces testing to cron jobs or manually triggered CI.
```

Comment:
Cypress dashboard service is not a publicly available page, I’d suggest giving some other example of subscriptions/users/organizations that people can relate to, e.g. Slack https://slack.com/intl/en-sk/help/articles/115004071768-What-is-Slack-#your-team-in-slack

Reply:
Added Slack to the example.